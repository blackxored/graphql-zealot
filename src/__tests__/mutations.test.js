import {
  addEdgeAndCursorToMutationResult,
  addEdgeToMutationResult,
  optimisticResponse,
} from '../mutations';

describe('addEdgeToMutationResult', () => {
  it('adds the object as the edge field on a payload', () => {
    const obj = { id: '1' };
    expect(addEdgeToMutationResult(obj)).toEqual({
      edge: {
        node: obj,
      },
    });
  });
});

describe('addEdgeAndCursorToMutationResult', () => {
  it('adds cursor and object as the edge field on a payload', async () => {
    const obj = 2;
    const connectionGetter = () => Promise.resolve([1, 2, 3, 4]);
    const result = await addEdgeAndCursorToMutationResult(connectionGetter)(2);

    expect(result).toHaveProperty('edge');
    expect(result.edge.cursor).toBeTruthy();
    expect(result.edge.node).toEqual(obj);
  });
});

describe('optimisticResponse', () => {
  it('adds an optimistic response shape to the given object', () => {
    const variables = {};
    const payload = {
      name: 'Foo',
    };

    expect(
      optimisticResponse('createUser', 'CreateUserPayload', payload, variables),
    ).toMatchSnapshot();
  });

  it('can accept a function to generate the response which is passed variables', () => {
    const variables = { id: 1 };
    const response = jest.fn(vars => ({
      ...vars,
      name: 'Foo',
    }));

    const result = optimisticResponse(
      'createUser',
      'CreateUserPayload',
      response,
      variables,
    );
    expect(result).toMatchSnapshot();
    expect(response).toHaveBeenCalledWith(variables);
  });
});
