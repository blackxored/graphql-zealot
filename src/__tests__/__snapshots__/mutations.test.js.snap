// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optimisticResponse adds an optimistic response shape to the given object 1`] = `
Object {
  "__typename": "Mutation",
  "createUser": Object {
    "__typename": "CreateUserPayload",
    "name": "Foo",
  },
}
`;

exports[`optimisticResponse can accept a function to generate the response which is passed variables 1`] = `
Object {
  "__typename": "Mutation",
  "createUser": Object {
    "__typename": "CreateUserPayload",
    "id": 1,
    "name": "Foo",
  },
}
`;
